UOP1_Project/GlobalSuppressions.cs

```diff
@@ -0,0 +1,70 @@
+[assembly: SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "<Pending>", Scope = "member", Target = "~P:GameInput.bindingMask")]
+[assembly: SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "<Pending>", Scope = "member", Target = "~P:GameInput.Gameplay")]
+[assembly: SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~P:GameInput.GameplayActions.enabled")]
+[assembly: SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "<Pending>", Scope = "member", Target = "~P:GameInput.Menus")]
+[assembly: SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~P:GameInput.MenusActions.enabled")]
+[assembly: SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "<Pending>", Scope = "member", Target = "~P:GameInput.MenusActions.enabled")]
+[assembly: SuppressMessage("StyleCop.CSharp.LayoutRules", "SA1513:Closing brace should be followed by blank line", Justification = "<Pending>", Scope = "member", Target = "~M:GameInput.GameplayActions.SetCallbacks(GameInput.IGameplayActions)")]
+[assembly: SuppressMessage("StyleCop.CSharp.LayoutRules", "SA1513:Closing brace should be followed by blank line", Justification = "<Pending>", Scope = "member", Target = "~M:GameInput.MenusActions.SetCallbacks(GameInput.IMenusActions)")]
+[assembly: SuppressMessage("StyleCop.CSharp.LayoutRules", "SA1513:Closing brace should be followed by blank line", Justification = "<Pending>", Scope = "type", Target = "~T:GameInput")]
+[assembly: SuppressMessage("Style", "IDE0044:Add readonly modifier", Justification = "<Pending>", Scope = "member", Target = "~F:GameInput.GameplayActions.m_Wrapper")]
+[assembly: SuppressMessage("Major Code Smell", "S2933:Fields that are only assigned in the constructor should be \"readonly\"", Justification = "<Pending>", Scope = "member", Target = "~F:GameInput.GameplayActions.m_Wrapper")]
+[assembly: SuppressMessage("Style", "IDE0044:Add readonly modifier", Justification = "<Pending>", Scope = "member", Target = "~F:GameInput.MenusActions.m_Wrapper")]
+[assembly: SuppressMessage("Major Code Smell", "S2933:Fields that are only assigned in the constructor should be \"readonly\"", Justification = "<Pending>", Scope = "member", Target = "~F:GameInput.MenusActions.m_Wrapper")]
+[assembly: SuppressMessage("StyleCop.CSharp.LayoutRules", "SA1515:Single-line comment should be preceded by blank line", Justification = "<Pending>", Scope = "member", Target = "~M:GameInput.#ctor")]
+[assembly: SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1214:Readonly fields should appear before non-readonly fields", Justification = "<Pending>", Scope = "member", Target = "~F:GameInput.m_Gameplay_Move")]
```

Comment:
This feels like it would be tough to maintain and more a specific way of ignoring errors rather then having a general setup. The Targets always correspond to the actual code unlike the .editorconfig which simply sets the rules for the whole project which would be a better way to go about this

Reply:
These ignores were necessary to add, because we don't want to update the auto-generated code. The reason is that, when it is generated again we will lost the work we've done to fix the errors.

Reply:
Hmmm... you have a valid point there. Would there be a cleaner way of doing this though? Instead of having a suppressing for each error type and location have the suppression for the entire file for all warnings? Otherwise this will become a beast to maintain which is not what we want

Reply:
StyleCop ignores autogenerated code, BUT files must be set as autogenerated in the right way:
`// <auto-generated/>`

Reply:
Resolved.

Reply:
Perfect! Then would this file itself still be required or can it be removed from the PR?